### PR TITLE
feat: update x/bank blocked addresses 

### PR DESCRIPTION
**Pull request by alexanderbez**
Update the deny-list provided to `x/bank` to allow `x/gov` and `x/groups` s.t. that they can RECEIVE funds :)

Blocked on:

- [ ] https://github.com/cosmos/cosmos-sdk/pull/11998
- [ ] https://github.com/cosmos/gaia/pull/1484

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -742,6 +742,21 @@ func (app *GaiaApp) ModuleAccountAddrs() map[string]bool {
 	return modAccAddrs
 }
 
+// BlockedModuleAccountAddrs returns all the app's blocked module account
+// addresses.
+func (app *GaiaApp) BlockedModuleAccountAddrs() map[string]bool {
```
Comment (glnro) [inline app/app.go:824]: Shouldn't we also replace
```go
app.BankKeeper = bankkeeper.NewBaseKeeper(
		appCodec,
		keys[banktypes.StoreKey],
		app.AccountKeeper,
		app.GetSubspace(banktypes.ModuleName),
		// app.ModuleAccountAddrs(),
                app.BlockedModuleAccountAddrs(),
	)
```
?

Comment (glnro) [reply app/app.go:824]: Tested this change locally with expected behavior 👍

Comment (yaruwangway) [reply app/app.go:824]: i think so

Comment (alexanderbez) [reply app/app.go:824]: Correct

File: app/app.go [hunk 2]
```diff
@@ -742,6 +742,20 @@ func (app *GaiaApp) ModuleAccountAddrs() map[string]bool {
...
+	modAccAddrs := app.ModuleAccountAddrs()
+
+	// remove module accounts that are ALLOWED to received funds
+	//
+	// TODO: Blocked on updating to v0.46.x
+	// delete(modAccAddrs, authtypes.NewModuleAddress(grouptypes.ModuleName).String())
```
Comment (yaruwangway) [inline app/app.go:830]: we can uncomment this and merge ? main branch is already in sdk 0.46.

Comment (glnro) [reply app/app.go:830]: Yes but I don't think the changes @alexanderbez made are in the release we're using, it's in `0.46.x` not `rc1`

Comment (glnro) [reply app/app.go:830]: https://github.com/cosmos/gaia/issues/1537

**Review by glnro — APPROVED** [review]

File: app/app.go (hunk 2 above)
Comment (yaruwangway) [reply app/app.go:830]: ok

**Pull request merged**